### PR TITLE
allow DatabricksSubmitRunOperator to accept a pipeline name for a pipeline_task

### LINKED ISSUE #32866: DatabricksSubmitRunOperator should accept a pipeline name for a pipeline_task
### Description

It would be nice if we could give the DatabricksSubmitRunOperator a pipeline name instead of a pipeline_id for cases when you do not already know the pipeline_id but do know the name. I'm not sure if there's an easy way to fetch a pipeline_id.

### Use case/motivation

Avoid hardcoding pipeline ID's, storing the ID's elsewhere, or fetching the pipeline list and filtering it manually if the pipeline name is known, but ID is not.

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by adam133**
closes: #32866

Allow running a databricks delta live tables pipeline by name without knowing the pipeline id.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by adam133** [comment]
Needs testing and tests. I just threw this together quickly because I had a bit to look at it.

**Comment by lafirm** [comment]
> Needs testing and tests. I just threw this together quickly because I had a bit to look at it. Will also need some documentation updates.

Hello @adam133, I'm very interested in contributing to this project, and I believe this PR would be an excellent place for me to get started. Would you mind if I join?

**Comment by adam133** [comment]
Should be good to review I think - let me know!

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Rebased to check - but i think there are still errors

**Comment by adam133** [comment]
Looks like I just needed to add the parameter name for the function call? Should be resolved.

**Comment by adam133** [comment]
Not sure how to get it to run the full tests again, but the stuff that was failing earlier should be fixed.

**Comment by adam133** [comment]
Ah ok looks like I'll need to do the `breeze` setup and figure out what it's doing...

**Comment by adam133** [comment]
Ok _now_ it should be good.

**Comment by adam133** [comment]
Ok I'm not sure what that latest error means. Does it have anything to do with my changes? Status = queued instead of success?

File: airflow/providers/databricks/hooks/databricks.py [hunk 1]
```diff
@@ -215,6 +216,67 @@ def find_job_id_by_name(self, job_name: str) -> int | None:
         else:
             return matching_jobs[0]["job_id"]
 
+    def list_pipelines(
+        self, max_results: int = 25, pipeline_name: str | None = None, notebook_path: str | None = None
+    ) -> list[dict[str, Any]]:
+        """
+        Lists the pipelines in Databricks Delta Live Tables.
+
+        :param max_results: The limit/batch size used to retrieve pipelines.
```
Comment (hussein-awala) [inline airflow/providers/databricks/hooks/databricks.py]: The name is confusing, I wonder if we can change it.

File: airflow/providers/databricks/operators/databricks.py [hunk 2]
```diff
@@ -384,6 +386,11 @@ def _get_hook(self, caller: str) -> DatabricksHook:
         )
 
     def execute(self, context: Context):
+        if "pipeline_task" in self.json and self.json["pipeline_task"].get("pipeline_id") is None:
```
Comment (hussein-awala) [inline airflow/providers/databricks/operators/databricks.py]: You need to add an extra check on `pipeline_name`:
```suggestion
        if (
            "pipeline_task" in self.json 
            and self.json["pipeline_task"].get("pipeline_id") is None
            and self.json["pipeline_task"].get("pipeline_name")
        ):
```

File: airflow/providers/databricks/hooks/databricks.py (hunk 1 above)
Comment (adam133) [reply airflow/providers/databricks/hooks/databricks.py]: Name of what exactly? Happy to change things, trying to align with how Databricks has them named.

Comment (potiuk) [reply airflow/providers/databricks/hooks/databricks.py]: Ywah looks consistent with the API

**Review by potiuk — APPROVED** [review]

File: airflow/providers/databricks/hooks/databricks.py (hunk 1 above)
Comment (hussein-awala) [reply airflow/providers/databricks/hooks/databricks.py]: Yes it is consistent with the API, but the API uses it to determine the total number of pipelines it should return by query, but in our case we collect all the pipelines, so IMHO `max_results_by_page` or `batch_size` are more suitable. 

It's a nit, if you don't agree you can merge.

Comment (potiuk) [reply airflow/providers/databricks/hooks/databricks.py]: I am happy with both

Comment (adam133) [reply airflow/providers/databricks/hooks/databricks.py]: Ah yeah I think that was a copy/paste from elsewhere. Switching it to `batch_size`. Attempting to run what tests I can locally then pushing it up.

**Pull request merged**